cli/flox/src/commands/activate.rs

```diff
@@ -417,182 +370,71 @@ impl Activate {
+            env: mode_link_path.to_string_lossy().to_string(),
+            env_project: concrete_environment.project_path()?,
+            env_cache: concrete_environment.cache_path()?.into_inner(),
+            env_description: now_active.bare_description(),
+            mode: mode.to_string(),
+            watchdog: (*WATCHDOG_BIN).clone(),
+            shell,
+            flox_active_environments: flox_active_environments.to_string(),
+            flox_env_log_dir: concrete_environment
+                .log_path()?
                 .to_string_lossy()
                 .to_string(),
-        );
-        command
-            .arg("--env-description")
-            .arg(now_active.bare_description());
+            prompt_color_1,
+            prompt_color_2,
+            flox_prompt_environments,
+            set_prompt,
+            flox_activate_store_path: store_path.to_string_lossy().to_string(),
+            // TODO: we should probably figure out a more consistent way to
+            // pass this since it's also passed for `flox build`
+            flox_runtime_dir: flox.runtime_dir.to_string_lossy().to_string(),
+            flox_services_to_start,
+            flox_env_cuda_detection,
+            flox_activate_start_services,
+            flox_services_socket,
+            interpreter_path,
+            invocation_type,
+            is_ephemeral,
+            run_args: self.run_args,
+        };
 
-        // Pass down the activation mode
-        command.arg("--mode").arg(mode.to_string());
+        let tempfile = tempfile::NamedTempFile::new_in(flox.temp_dir)?;
 
-        command
-            .arg("--watchdog")
-            .arg(WATCHDOG_BIN.to_string_lossy().to_string());
-
-        // when output is not a tty, and no command is provided
-        // we just print an activation script to stdout
-        //
-        // That script can then be `eval`ed in the current shell,
-        // e.g. in a .bashrc or .zshrc file:
-        //
-        //    eval "$(flox activate)"
-        if invocation_type == InvocationType::InPlace {
-            let shell = Self::detect_shell_for_in_place()?;
-            subcommand_metric!("activate", "shell" = shell.to_string());
-            command.arg("--shell").arg(shell.exe_path());
-            Self::activate_in_place(command, shell)?;
-
-            return Ok(());
-        }
+        let writer = BufWriter::new(&tempfile);
+        serde_json::to_writer_pretty(writer, &activate_data)?;
 
-        let shell = Self::detect_shell_for_subshell();
-        subcommand_metric!("activate", "shell" = shell.to_string());
-        command.arg("--shell").arg(shell.exe_path());
-        // These functions will only return if exec fails
-        if invocation_type == InvocationType::Interactive {
-            Self::activate_interactive(command)
-        } else {
-            Self::activate_command(command, self.run_args, is_ephemeral)
-        }
-    }
+        let flox_activations = env!("FLOX_ACTIVATIONS_BIN");
```

Comment:
Question: does this need to be handled like the other `*_BIN` variables e.g. devtime/runtime overrideable?